### ISSUE TITLE
Multiline if-else does not format `else if`

### ISSUE THREAD

**Issue by pdb-stripe** [pinterest/ktlint#2420]
I observed yesterday that when enabling the multiline-if-else rule and running the formatter, `else-if` blocks were not converted to multiline.

## Expected Behavior
Given this code:

````kotlin
if (foo) doFoo()
else if (bar) doBar()
````

the formatter for the `multiline-if-else` rule should convert it to:

````kotlin
if (foo) {
    doFoo()
} else if (bar) {
    doBar()
}
````

## Observed Behavior
This code actually formats to:

````kotlin
if (foo) {
    doFoo()
} else if (bar) doBar()
````

## Steps to Reproduce
We had code with this syntax:

````kotlin
if (foo) return doFoo()
else if (bar) return doBar()
return null
````

Since it probably should have used a lifted return statement or a when statement (or both), my guess is this unique scenario would not be discovered if some other rules were not also disabled - but not entirely sure.

## Your Environment
Ktlint 0.50.0, MacOS

This was executed using ktlint-gradle (via `./gradlew ktlintFormat`) 11.6.1 on Gradle 8.5
